tikv/raftstore/fsm_peer.go

```diff
@@ -920,113 +917,590 @@ func (d *peerFsmDelegate) onStaleMerge() {
 }
 
 func (d *peerFsmDelegate) onReadyApplySnapshot(applyResult *ApplySnapResult) {
-	// TODO: stub
+	prevRegion := applyResult.PrevRegion
+	region := applyResult.Region
+
+	log.Infof("%s snapshot for region %s is applied", d.tag(), region)
+	d.ctx.storeMetaLock.Lock()
+	defer d.ctx.storeMetaLock.Unlock()
+	meta := d.ctx.storeMeta
+	initialized := len(prevRegion.Peers) > 0
+	if initialized {
+		log.Infof("%s region changed from %s -> %s after applying snapshot", d.tag(), prevRegion, region)
+		meta.regionRanges.Delete(prevRegion.EndKey)
```

Comment:
Should check the deleted region's id?

Reply:
Then we have to pay the cost of an extra `Get`.
I assume we check it because the `remove` function in TiKV returns the removed item.